tests/examples/utils.py

```diff
@@ -0,0 +1,137 @@
+import itertools
+import re
+import shlex
+from pathlib import Path
+from subprocess import CompletedProcess
+from typing import Generator, Iterable, List, Optional, TypeVar, Union
+
+import pytest
+from bs4 import BeautifulSoup, ResultSet, Tag
+from cmarkgfm import github_flavored_markdown_to_html as gfm_to_html
+
+from tests.testing_utils import is_gpu_available, is_torch_available
+
+_T = TypeVar("_T")
+
+requires_gpu = pytest.mark.skipif(not is_gpu_available(), reason="GPU is required")
```

Comment:
what is this marking?

Reply:
It gets imported in the test files and is used throughout as a decorator for all tests that require a GPU.

Reply:
Sort of, yes. This is a pytest-compatible version of the existing decorator which is a very thin wrapper over the core logic that the other one is based on (`is_gpu_available()`), and it’s generally limited to the examples tests. As long as that underlying function exists and returns a boolean, this wrapper will not need any changes.

Unfortunately, to use the other existing “requires gpu” wrapper, I would have to swap to using `unittest` to compose the tests, adding a lot of overhead that pytest provides for free (namely both their built-in fixtures such as `tmp_path` as well as their ability to very easily create simple fixtures and other test decorators).

Alternatively, I did create a separate, more granular pytest skip decorator that checks for a specific number of GPUs and skips if there are fewer available. If you’d prefer, can drop the pytest version of `requires_gpu` and replace all of its usages with that decorator (e.g. `requires_gpu_count(1)`).

Let me know how you’d like me to rectify this.

Reply:
I have taken the latter approach — my pytest-compatible `requires_gpu` skip decorator has been resolved, and all uses of it have been replaced with `requires_gpu_count(1)`.